images/converter/converter.go

```diff
@@ -24,49 +24,63 @@ import (
 	"github.com/containerd/containerd/images"
 	"github.com/containerd/containerd/leases"
 	"github.com/containerd/containerd/platforms"
+	"golang.org/x/sync/singleflight"
 )
 
 type convertOpts struct {
 	layerConvertFunc ConvertFunc
 	docker2oci       bool
 	indexConvertFunc ConvertFunc
 	platformMC       platforms.MatchComparer
+	hooks            ConvertHooks
+	sg               *singleflight.Group
 }
 
 // Opt is an option for Convert()
-type Opt func(*convertOpts) error
+type Opt func(*convertOpts)
```

Comment:
Ok, just want to keep `DefaultIndexConvertFunc` and `IndexConvertFuncWithHook` with no `error` return, I'll think it over.

Reply:
@ktock Added the error return for `DefaultIndexConvertFunc(...)` and `IndexConvertFuncWithHook`, I worry that it will require code changes to stargz-snapshotter or other components used, PTAL.

Reply:
Updated: fix a possible diff id map conflict in the third commit.